PowerShell Scanners/Battery Status/Battery Status.ps1

```diff
@@ -1,23 +1,33 @@
-﻿function Format-Battery($BatteryName = "No Battery Detected", $DesignedCapacity = 0, $FullCharge = 0, $Health = 0) {
+﻿$Batteries = (Get-CimInstance -ClassName "BatteryStatus" -Namespace "ROOT\WMI" -ErrorAction SilentlyContinue)
+$AllBatteryData = Get-CimInstance -ClassName "BatteryStaticData" -Namespace "ROOT\WMI"
```

Comment:
I get a `Get-CimInstance: Generic failure` on this line

Reply:
Does the old script work for you?

Reply:
Yes it did.

Reply:
function Format-Battery($BatteryName = "No Battery Detected", $DesignedCapacity = 0, $FullCharge = 0, $Health = 0) {
    [PSCustomObject]@{
        Name             = $BatteryName
        DesignedCapacity = $DesignedCapacity
        FullCharge       = $FullCharge
        Health           = $Health
    }
}


$Batteries = (Get-WmiObject -Class "BatteryStatus" -Namespace "ROOT\WMI" -ErrorAction SilentlyContinue)
$AllBatteryData = Get-WmiObject -Class "BatteryStaticData" -Namespace "ROOT\WMI"
$AllCharges = (Get-WmiObject -Class "BatteryFullChargedCapacity" -Namespace "ROOT\WMI").FullChargedCapacity
$BatteryIndex = 0
foreach ($Battery in $Batteries) {
    $BatteryName = $AllBatteryData.DeviceName.split(" ")[$BatteryIndex]
    $DesignedCapacity = $AllBatteryData.DesignedCapacity[$BatteryIndex]
    $FullCharge = $AllCharges[$BatteryIndex]
    $Health = [math]::Round($FullCharge / $DesignedCapacity * 100, 2)

    Format-Battery $BatteryName $DesignedCapacity $FullCharge $Health
    $BatteryIndex++
}

returns this for me from original post

Name DesignedCapacity FullCharge Health
---- ---------------- ---------- ------
DELL            60002      55054  91.75

Reply:
I don't know why, but `Get-CimInstance` just doesn't work with `BatteryStaticData`. I changed it back to `Get-WmiObject`. Also, I installed Windows on a laptop and this script is working on it now. I should have done that the first time :)